src/__checks__/table-index.check.ts

```diff
@@ -0,0 +1,47 @@
+import { defineDb, defineTable, integer, text, timestampWithTimeZone, uuid } from '..';
+import { expect, describe, test } from 'tstyche';
+import { btree, gin, gist } from '../table-index-types';
+import { Index } from '../table-index';
+
+const foo = defineTable({
+  columns: {
```

Comment:
Would prefer to avoid nesting the columns under `columns`. What about an optional second parameter, e.g.

```ts
const foo = defineTable({
    ...
}, {
    indexes: t => ({
        fooPkey: btree(t.id).primaryKey(),
    })
});
```

I think it might be worth using the pattern `t.id` instead of `foo.id` because it's more concise.

Reply:
We can definitely change `defineTable()` to take in the columns and the index definitions as separate parameters. I think we still want to refactor `TableDefinition` to have the columns under `columns`, though, since we need the column and index definitions to be distinguished somehow in the single `TableDefinition` object.

`defineTable()`, then, will just create a `TableDefinition` from a `ColumnsDefinition` and a `defineIndexes()`. I'll go ahead and make the change.

Reply:
I liked `foo.id` to reinforce that the parameter is basically the `foo` table, but I'm cool with either - I'll change it.

Reply:
In this context, we know it's all one table, so the `foo.` feels like almost pure noise.